bin/cisc_to_adie_conversion.py

```diff
@@ -0,0 +1,82 @@
+# Script to convert CISC ID to ADIE ID in directory and file names 
+# This is only tested on directories/file£s with a BIDS naming convention
+# The way the script searches for the sub- ID is dependent on the characters before and after,
+# which is specific to BIDS - be aware  
+
+import pandas as pd
+import os 
+import re
+import shutil
+import sys, os
+
+'''
+# Construct path of converter .txt file such that it's relative to the user running the script
+# Get parent directory of this script       
+pathname = os.path.dirname(sys.argv[0])  
+# Get the level above that (i.e. the critchley_adie project dir)      
+project_path = os.path.split(pathname)[0]
+print('project_path=',project_path)
+'''
+
+#TODO: make this path relative to the user 
+#Import adie/cisc conversion txt file and store as dataframe
+txt=("/Volumes/cisc2/projects/critchley_adie/BIDS_data/sourcedata/adie_idconvert.txt")
+
+# convert this to dictonary, where key = CISC and val = ADIE 
+rename = {}
+with open(txt) as f:
+    for line in f:
+        (key,val)=line.split()
+        rename[str(key)] = str(val)
+
+
+# USE OS.WALK
+
+def subconvert(p):
+# P should = the path of the directory above the sub- dirs 
+    for root,dirs,files in os.walk(p):
+        for f in files:
+            # Extract sub- number for searching 
+            # Extract number after 'sub-' and before '_'
```

Comment:
We can improve the current search pattern to "sub-([0-9]*)". String the bracket will be recognised as a group.
I suggest using a dictionary so you are searching for the exact match.
Take the file name for example:
```python
# dictionary entry - CISCID : ADIEIDE 
# strip off CISC when saving to a dict
conversion = {"34233": "ADIE999", "54364": "ADIE998" .... }

cisc_bidsname = "sub-34233_stuff.nii.gz"
cur_subid = re.search("sub-([0-9]*)", cisc_bidsname).groups()[0]
adie_bidsname = cisc_bidsname.replace(cur_subid, conversion[cursubid])
```